### PR TITLE
[test] Implement rom_e2e_epmp_init

### LINKED ISSUE #14497: [rom-e2e] rom_e2e_epmp_init
**Testpoint name:** [rom_e2e_epmp_init](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson?q=rom_e2e_epmp_init)

**Contact person:** @alphan

**Description:** Verify that ROM correctly configures the ePMP region for debug ROM.

- ePMP Debug ROM region should be enabled in TEST, DEV, and RMA, anddisabled in PROD and
  PROD_END.

### PR DESCRIPTION
**Pull request by dmcardle**
Fixes #14497

Signed-off-by: Dan McArdle <dmcardle@google.com>

### CODE REVIEW

**Review by alphan — COMMENTED** [review]
Thanks @dmcardle!

I think it would be better to implement this as a flash program (using `opentitan_functest`) instead of a gdb test. Also, the test should check for the ePMP config so, it should let the rom execute at least until after `rom_epmp_config_debug_rom()` is called.

**Comment by dmcardle** [comment]
> Thanks @dmcardle!
> 
> I think it would be better to implement this as a flash program (using `opentitan_functest`) instead of a gdb test. Also, the test should check for the ePMP config so, it should let the rom execute at least until after `rom_epmp_config_debug_rom()` is called.

I took a stab at this, PTAL! WDYT about determining whether debugging is enabled inside the test? Alternatively, I could have Bazel set a preprocessor definition.

I've only tested the cw310 tests, FYI. Currently running the Verilator tests.

**Comment by dmcardle** [comment]
(Verilator tests passed.)

**Comment by dmcardle** [comment]
@alphan Ping :)

File: sw/device/silicon_creator/rom/e2e/epmp_init_test.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static bool is_debug_enabled() {
+  dif_lc_ctrl_t lc_ctrl;
+  mmio_region_t lc_reg = mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR);
+  CHECK_DIF_OK(dif_lc_ctrl_init(lc_reg, &lc_ctrl));
+  return lc_ctrl_testutils_debug_func_enabled(&lc_ctrl);
+}
```
Comment (alphan) [inline sw/device/silicon_creator/rom/e2e/epmp_init_test.c:26]: Let's don't trust the test code for this. You can approach this multiple ways but one option could be just print out the config and set the success/fail strings according to the lifecycle state.

Comment (dmcardle) [reply sw/device/silicon_creator/rom/e2e/epmp_init_test.c]: OK, I opted to inject a preprocessor macro that tells the test whether to expect debugging to be enabled. WDYT?

**Review by alphan — APPROVED** [review]

**Pull request merged**